packages/discord.js/src/structures/ModalBuilder.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+const { ModalBuilder: BuildersModal, isJSONEncodable } = require('@discordjs/builders');
+const Transformers = require('../util/Transformers');
+
+/**
+ * Represents a modal builder.
+ */
+class ModalBuilder extends BuildersModal {
+  constructor({ components, ...data }) {
+    super({
+      components: components?.map(c => (isJSONEncodable(c) ? c.toJSON() : Transformers.toSnakeCase(c))),
```

Comment:
shouldn't this do an `instanceof Component` check instead like `ActionRowBuilder`'s constructor?

Reply:
no, I recommended in discord to use `isJSONEncodable`, action row should do the same at some point.

Reply:
We decided to go with the `instanceof` operator as to not lose references to component classes.